cmd/format.go

```diff
@@ -91,16 +91,16 @@ func randSeq(n int) string {
 
 func doTesting(store object.ObjectStorage, key string, data []byte) error {
 	if err := store.Put(key, bytes.NewReader(data)); err != nil {
-		if strings.Contains(err.Error(), "Access Denied") {
-			return fmt.Errorf("Failed to put: %s", err)
-		}
-		if err2 := store.Create(); err2 != nil {
-			return fmt.Errorf("Failed to create %s: %s,  previous error: %s\nplease create bucket %s manually, then format again",
-				store, err2, err, store)
-		}
-		if err := store.Put(key, bytes.NewReader(data)); err != nil {
-			return fmt.Errorf("Failed to put: %s", err)
+		if strings.Contains(err.Error(), "NoSuchBucket") {
```

Comment:
there is no guarantee that all object stores are using this error message

Reply:
Yes, that is a reasonable assumption. But most object stores follow the S3 API, so this conditional logic should be able to handle most cases.

Reply:
If some object stores don't follow this convention it doesn't have a big impact on this logic.

Reply:
I've updated the description of this PR to explain why this fix is necessary.

Reply:
After this change, we will not try to create the bucket if different message returned, for example ceph, we will not create the pool.

Reply:
How much of an impact would it have if the bucket could not be created automatically?